cmd/clusterctl/cmd/root.go

```diff
@@ -57,7 +57,9 @@ func Execute() {
 				}
 			}
 		}
-		// TODO: print cmd help if validation error
+		if RootCmd.CalledAs() != "clusterctl" {
+			RootCmd.Help()
+		}
```

Comment:
i think the origin comment implies that validation should be added here too.

/assign @fabriziopandini 
https://github.com/kubernetes-sigs/cluster-api/commit/74c8cf37de8507f98a977fb2eff2faced72928dc

Reply:
I pondered this too initially. When the code gets to line 60, we already know it's an error. What additional information do you think we need at this point to justify calling `RootCmd.Help()`?

I suppose one thing about getting more information about the error could perhaps allow us to print the help message in relation to a specific command. For example, if the user executes:

`$ clusterctl move --something-invalid`

we could show the command help you'd otherwise see by running:

`$ clusterctl move --help`

Reply:
> i think the origin comment implies that validation should be added here too.

I may have missed your point here. Will *all* other kinds of errors lead here too?

Reply:
If I remember well the original intent of this comment, was to provide the help of the command only if the error was due to parameter validation (sorry if I did not provide more content). 

However, AFAIK:

1. in clusterctl there is not yet a clear definition of what is a validation error, nor a way to distinguish validation errors from processing errors
2. validation happens both in the `/cmd` and in `/client`

So, IMO, the long term fix for this is

- To define how 1 should be addressed
- To adapt `/client` code to the new definition
- To move as much as possible of `/cmd` validation into `/client`, so the library UX will be more consistent with CLI UX (when this makes sense)
- To adapt remaining `/cmd` validations to the new definition

WRT to the change proposed in this doc, given the additional info above, IMO this is more a workaround than a fix to the original problem; said that I have two concerns on the proposed implementation:

- I fear that, in case of processing errors, printing the help makes the output verbose without providing information useful to the users (parameters are already ok, so why to print the command help again?)
- I don't want to lose track of the original intent by simply dropping the TODO, even if this is not a priority but a nice to have

Reply:
> WRT to the change proposed in this doc

I agree, this proposed change is just a workaround. We need some way of being able to determine the kind of error.

As this seems like it'll require a bit more work, is it work me closing this PR and creating a new issue including the details commented on here?

Reply:
I'm ok to close.
Feel free to move above discussion on an issue if you feel this is required

Reply:
Closing as per the above discussion. @hazbo please create a new issue if needed - thanks!